python/tests/integration/arcticdb/test_read_batch_more.py

```diff
@@ -0,0 +1,364 @@
+                                    ReadRequest(symbol2, date_range=date_range),
+                                    ReadRequest(symbol2, date_range=date_range1, columns=columns_to_select1, as_of=0)
+                                    ])
+    
+    
+    assert [vi.symbol for vi in batch] == [symbol3, symbol1, symbol1, symbol1, symbol2, symbol2, symbol2, symbol2]
+    assert isinstance(batch[0], DataError)
+    assert batch[0].symbol == symbol3
+    assert_frame_equal(df1_all, batch[1].data)
+    assert_frame_equal(df1_till3, batch[2].data)
+    assert_frame_equal(df1_0, batch[3].data)
+    assert batch[4].data.empty
+    # Only 3 colums in the DF, no date filtering
+    assert_frame_equal(df2_all_col_filtered, batch[5].data)
+    # Check if datetime filtering first several and last several is ok
+    assert_frame_equal(df2_all_without_first_and_last, batch[6].data)
+    # Column filters + datetime filters applied on the result
+    assert_frame_equal(df2_0_allfilters, batch[7].data)
+
+@pytest.mark.xfail(reason = "ArcticDB#1970")
+def test_read_batch_query_with_and(arctic_library):
+    """
+        A very small test to isolate the problem with usage of "and" 
+        in arctic queries. It produces wrong result, and should have 
+        raised an error
+    """
+
+    lib = arctic_library
+
+    symbol = "_s_"
+    df = get_sample_dataframe(100)
+
+    dfq = "bool == True and int8 > 5"
+    q = QueryBuilder()
+    q = q[q["bool"] and (q["int8"] > 5)]
+
+    lib.write(symbol, df)
+
+    batch = lib.read_batch(symbols=[ReadRequest(symbol, query_builder=q)])
+    df_result = df.query(dfq, inplace=False)
+
+    assert batch[0].symbol == symbol
+    assert isinstance(batch[0], DataError)
+
+def test_read_batch_metadata_on_different_version(arctic_library):
+    """
+        Here we test if read of metadata over several different states of DB with
+        several differen read_batch() invokations works correctly.
+        Thus we check isolation of the method over times
+    """
+
+    lib = arctic_library
+
+    symbol = "_s_"
+    df_0 = get_sample_dataframe(1)
+    df_1 = get_sample_dataframe(2, seed=100)
+    df_2 = get_sample_dataframe(3, seed=1345)
+    df_3 = get_sample_dataframe(4, seed=1345)
+    meta0 = {"meta0" : 0, "a" : "b", "c" : 1, 2 : 3}
+    meta1 = {"meta1" : 1, "arr" : [1, 2, 4]}
+    meta2 = {"meta2" : 2, 1 : {}, "arr2" : [1, 2, 4]}
+    df_till1 = pd.concat([df_0, df_1])
+    df_all = pd.concat([df_till1, df_2, df_3])
+
+    lib.write(symbol, df_0, metadata=meta0)
+    lib.append(symbol, df_1)
+    lib.write_metadata(symbol, meta1)
+
+    batch = lib.read_batch(symbols=[ReadRequest(symbol, as_of=2),
+                                    ReadRequest(symbol, as_of=0),
+                                    ReadRequest(symbol, as_of=1)])
+    
+    assert meta1 == lib.read_metadata(symbol).metadata
+    assert meta0 == batch[1].metadata
+    assert meta1 == batch[0].metadata
+    assert None == batch[2].metadata
```

Comment:
In Python `None` should never be tested with `==` you should use `batch[2].metadata is None`. Please fix other occurrences below.

Reply:
Thanks! done!